command/agent/config/config.go

```diff
@@ -22,6 +22,17 @@ type Config struct {
 	AutoAuth      *AutoAuth `hcl:"auto_auth"`
 	ExitAfterAuth bool      `hcl:"exit_after_auth"`
 	PidFile       string    `hcl:"pid_file"`
+	Cache         *Cache    `hcl:"cache"`
+}
+
+type Cache struct {
+	UseAutoAuthToken bool        `hcl:"use_auto_auth_token"`
+	Listeners        []*Listener `hcl:"listeners"`
+}
+
+type Listener struct {
```

Comment:
Can you instead just use the same Listener that we use in Vault? I think the parsing code below is basically a straight copy so it seems like a win to factor it out somewhere (e.g. helper/listener or so).